freeipa/src/main/java/com/sequenceiq/freeipa/service/freeipa/user/model/UmsUsersState.java

```diff
@@ -0,0 +1,59 @@
+package com.sequenceiq.freeipa.service.freeipa.user.model;
+
+import static java.util.Objects.requireNonNull;
+
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+
+public class UmsUsersState {
+
+    private final UsersState usersState;
+
+    private final Map<String, WorkloadCredential> usersWorkloadCredentialMap;
+
+    private final Set<FmsUser> requestedWorkloadUsers;
+
+    public UmsUsersState(UsersState usersState, Map<String, WorkloadCredential> usersWorkloadCredentialMap, Set<FmsUser> requestedWorkloadUsers) {
+        this.usersState = requireNonNull(usersState, "UsersState is null");
+        this.usersWorkloadCredentialMap = requireNonNull(usersWorkloadCredentialMap, "workload credential map is null");
+        this.requestedWorkloadUsers = requireNonNull(requestedWorkloadUsers, "requested workload users is null");
```

Comment:
I would make a copy of the map and set, even if you don't make the members immutable.